src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -279,7 +273,7 @@ public static class DependencyGraphRestoreUtility
             cacheContext.SolutionSpec = solutionDgSpec;
             cacheContext.SolutionSpecHash = newDependencyGraphSpecHash;
 
-            if (oldDependencyGraphSpecHash != newDependencyGraphSpecHash)
+            if (forceRestore || (oldDependencyGraphSpecHash != newDependencyGraphSpecHash))
             {
```

Comment:
This seems correct. Force is only done during a rebuild, all of the work done here to build the dg file is stored in the cache context and used again later on. The time different should only be the time it takes to create the hash, which @dtivel has perf numbers on.